control/controlhttp/constants.go

```diff
@@ -89,6 +90,8 @@ type Dialer struct {
 	drainFinished        chan struct{}
 	omitCertErrorLogging bool
 	testFallbackDelay    time.Duration
+
+	Clock tstime.Clock
```

Comment:
all the other exported fields are up above the unexported ones and have docs including the behavior on a zero value. Can this be nil? What does that mean? It should probably mean stdclock.